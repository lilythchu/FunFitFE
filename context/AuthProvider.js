import React, {createContext, useContext, useState} from 'react';

const AuthContext = createContext();

const AuthProvider = ({children}) => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [profile, setProfile] = useState({});
  const [token, setToken] = useState();

  return (
    <AuthContext.Provider
      value={{
        isLoggedIn,
        setIsLoggedIn,
        profile,
        setProfile,
        token,
        setToken,
      }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useLogin = () => useContext(AuthContext);

export default AuthProvider;
