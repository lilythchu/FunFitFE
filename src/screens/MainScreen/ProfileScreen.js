import { StyleSheet, Text, View, Button } from 'react-native'
import React from 'react'
import { globalStyles } from '../../../styles/global.js'
import { useNavigation } from '@react-navigation/native'
import { useLogin } from '../../../context/AuthProvider.js'

const ProfileScreen = () => {
  const {setIsLoggedIn} = useLogin();
  const navigation = useNavigation();
  const handleLogOut = () => {
    setIsLoggedIn(false);
  }

  return (
    <View style={globalStyles.container}>
      <Text style={globalStyles.paragraph}>ProfileScreen</Text>
      <Button title='Log out' onPress={handleLogOut}/>
    </View>
  )
}

export default ProfileScreen

const styles = StyleSheet.create({})