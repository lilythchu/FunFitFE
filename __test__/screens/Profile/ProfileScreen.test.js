import React from 'react';
import {render, screen} from '@testing-library/react-native';
import AuthProvider from '../../../context/AuthProvider';
import ProfileScreen from '../../../src/screens/Core/Profile/ProfileScreen';

const mockedNavigate = jest.fn();
const mockInfo = {
  name: 'friendName',
  workoutInterests: ['yoga'],
  age: '20',
  points: 0,
  level: 1,
}

describe('Profile screen', () => {
  test('renders correctly', () => {
    render(
      <AuthProvider>
        <ProfileScreen/>
      </AuthProvider>
    );
    expect(screen).toMatchSnapshot();
  })
})