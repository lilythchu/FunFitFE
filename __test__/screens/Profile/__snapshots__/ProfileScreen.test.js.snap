// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile screen renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "white",
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }
    >
      <ActivityIndicator
        size="large"
        style={
          Object {
            "color": "black",
            "padding": 25,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "paddingBottom": 5,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#666",
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRightColor": "grey",
            "borderRightWidth": 1,
            "flex": 6,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
           Level 
        </Text>
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 16,
              "fontWeight": "600",
              "marginBottom": 10,
              "textAlign": "center",
            }
          }
        >
           
           
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 6,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
           Points 
        </Text>
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 16,
              "fontWeight": "600",
              "marginBottom": 10,
              "textAlign": "center",
            }
          }
        >
           
           
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f2f1f0",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            Settings
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
            >
              Account Settings
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Object {
            "maxHeight": "0%",
            "opacity": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#f2f1f0",
              "borderRadius": 20,
              "marginHorizontal": 30,
              "padding": 10,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginVertical": 4,
                      "paddingRight": 4,
                    }
                  }
                >
                  <Text>
                    Name
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholderTextColor="#86939e"
                  style={
                    Object {
                      "color": "#242424",
                      "flex": 1,
                      "fontSize": 15,
                      "minHeight": 40,
                      "textAlign": "center",
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginVertical": 4,
                      "paddingRight": 4,
                    }
                  }
                >
                  <Text>
                    Bio
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholderTextColor="#86939e"
                  style={
                    Object {
                      "color": "#242424",
                      "flex": 1,
                      "fontSize": 15,
                      "minHeight": 40,
                      "textAlign": "center",
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginVertical": 4,
                      "paddingRight": 4,
                    }
                  }
                >
                  <Text>
                    Age
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  keyboardType="numeric"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder=""
                  placeholderTextColor="#86939e"
                  style={
                    Object {
                      "color": "#242424",
                      "flex": 1,
                      "fontSize": 15,
                      "minHeight": 40,
                      "textAlign": "center",
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                Object {
                  "marginTop": 10,
                },
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#68bbe3",
                    "borderColor": "transparent",
                    "borderRadius": 30,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginBottom": 40,
                    "marginHorizontal": 20,
                    "marginVertical": 20,
                    "padding": 8,
                    "width": 375,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Update
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
            >
              Change workout interests
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transparent={true}
        visible={false}
      />
    </View>
    <View
      style={
        Object {
          "marginVertical": 10,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f2f1f0",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            More
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
            >
              About us
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingLeft": 16,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transparent={true}
        visible={false}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#bcbbc1",
            "flexDirection": "row",
            "marginBottom": 50,
            "padding": 14,
          }
        }
        testID="padView"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            Log out
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
